sdk/java-sdk-spring/src/main/scala/kalix/javasdk/impl/Validations.scala

```diff
@@ -161,6 +167,59 @@ object Validations {
     }
   }
 
+  private def getEventType(esEntity: Class[_]): Class[_] = {
+    val genericTypeArguments = esEntity.getGenericSuperclass
+      .asInstanceOf[ParameterizedType]
+      .getActualTypeArguments
+    genericTypeArguments(1).asInstanceOf[Class[_]]
+  }
+
+  private def missingEventHandlerValidations(
+      component: Class[_],
+      updateMethodPredicate: Method => Boolean): Validation = {
+    val methods = component.getMethods.toIndexedSeq
+
+    if (hasEventSourcedEntitySubscription(component)) {
+      eventSourcedEntitySubscription(component)
+        .map { classLevel =>
+          val eventType = getEventType(classLevel.value())
+          if (!classLevel.ignoreUnknown() && eventType.isSealed) {
+            val effectMethodsInputParams: Seq[Class[_]] = methods
+              .filter(updateMethodPredicate)
+              .map(_.getParameterTypes.last) //last because it could be a view update methods with 2 params
+            val errors = eventType.getPermittedSubclasses
+              .filterNot(effectMethodsInputParams.contains)
+              .map(clazz => s"Missing event handler for ${clazz.getName}")
+            Validation(errors)
+          } else {
+            Valid
+          }
+        }
+        .getOrElse(Valid)
+    } else {
+      //method level
+      val effectOutputMethodsGrouped = methods
+        .filter(hasEventSourcedEntitySubscription)
+        .filter(updateMethodPredicate)
+        .groupBy(findEventSourcedEntityClass)
+
+      val errors = effectOutputMethodsGrouped.flatMap { case (entityClass, methods) =>
+        val eventType = getEventType(entityClass)
+        if (eventType.isSealed) {
+          val effectOutputInputParams: Seq[Class[_]] = methods.map(_.getParameterTypes.last)
+          eventType.getPermittedSubclasses
+            .filterNot(effectOutputInputParams.contains)
+            .map(clazz =>
+              s"Component ${component.getSimpleName} missing event handler for ${clazz.getName} from ${entityClass.getSimpleName}")
```

Comment:
```suggestion
              s"Component '${component.getSimpleName}' is missing event handler for '${clazz.getName}' from '${entityClass.getSimpleName}'")
```

Reply:
Maybe the last part: "from `entityClass.getSimpleName`" can be removed?

Reply:
ok